### ISSUE TITLE
rewrite lint rule decoration feature based on lix validation rules

### ISSUE THREAD

**Issue by felixhaeberle** [opral/inlang-cli#15]
**Description**

@samuel.stroschein has made a decision to prematurely remove lint rule functionality from the inlang sdk and move the functionality/responsibility to lix, calling it "lix validation rules". <br><br>This decision leads to Sherlock not being able to offer lint rules anymore (unless we have shipped Lix validation rules) which not only hurts our users in the meantime of a prerelease but also adds double or tripple the work – for negative value. <br><br>On the other hand, Lix validation rules could be a great concept but is currently not spec'ed out, nobody is working at it currently and it seems that it has low priority, seemingly lower than building a host.<br><br>**Proposal**

Revert the decision, build lix validation rules & only after release Sherlock on SDK v2. Everything else results in work done thrice & damaging user trust.

**Comment by samuelstroschein** [comment]
Disagree. Release Sherlock. Validation rules take too long. 

I am okay with shipping utility functions that simulate validation rules. See [MESDK-204](https://linear.app/opral/issue/MESDK-204/missingtranslations-query-utility). 

@felix.haeberle what are the rules you are missing? Missing translations seems to be the only crucial one

**Comment by felixhaeberle** [comment]
forced concensus: we will ship lint rules as utility functions. therefore build utility functions in the sdk. later change it to lix validation rules. creates thrice the effort & work not done at the right place but unblocks sherlock & fink fn. cc @niklas.buchfink

**Comment by samuelstroschein** [comment]
i agree with the consensus (but disagree with your wording. the effort of utility functions is trivial compared to lix validation rules)

**Comment by felixhaeberle** [comment]
progress can be tracked in [SHERL-113](https://linear.app/opral/issue/SHERL-113/integrate-lint-rules-in-vscode-diagnostics-and-inline-display-feature) – closing this.
